pkg/koordlet/util/system/kidled_util.go

```diff
@@ -128,17 +136,38 @@ func ParseMemoryIdlePageStats(content string) (*ColdPageInfoByKidled, error) {
 	return &info, nil
 }
 
+// boundary is the index of [1,2)  [2,5)  [5,15)  [15,30)  [30,60)  [60,120)  [120,240)  [240,+inf).
+// if boundary is equal to 3, it will compute sum([5*scan_period_scands,+inf)) of cold page cache
 func (i *ColdPageInfoByKidled) GetColdPageTotalBytes() uint64 {
 	sum := func(nums ...[]uint64) uint64 {
 		var total uint64
 		for _, v := range nums {
-			for _, num := range v {
-				total += num
+			for i, num := range v {
+				if i >= int(kidledColdBoundary) {
+					total += num
+				}
 			}
 		}
 		return total
 	}
-	return sum(i.Csei, i.Dsei, i.Cfei, i.Dfei, i.Csui, i.Dsui, i.Cfui, i.Dfui, i.Csea, i.Dsea, i.Cfea, i.Dfea, i.Csua, i.Dsua, i.Cfua, i.Dfua, i.Slab)
+	return sum(i.Cfei, i.Dfei, i.Cfui, i.Dfui)
+}
+
+// boundary is the index of [1,2)  [2,5)  [5,15)  [15,30)  [30,60)  [60,120)  [120,240)  [240,+inf).
+// if boundary is equal to 3, it will compute sum([5*scan_period_scands,+inf)) of hot page cache
+func (i *ColdPageInfoByKidled) GetHotPageTotalBytes() uint64 {
+	sum := func(nums ...[]uint64) uint64 {
```

Comment:
1. use a function instead of a closure
2. the inner loop is a suboptimal implementation